### PR TITLE
Change startTime_ to endTime_ in FutureInfo

### PR DESCRIPTION
**Pull request by xush6528**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #31078 Change type of timeoutFutures_ key to time_point instead of duration
* **#30342 Change startTime_ to endTime_ in FutureInfo**

This can eliminate the unnecessary calls to getRPCEndTime(). Reduce lines of code for simplicity.

Differential Revision: [D5705624](https://our.internmc.facebook.com/intern/diff/D5705624/)

### CODE REVIEW

File: torch/csrc/distributed/rpc/process_group_agent.h [hunk 1]
```diff
@@ -81,16 +81,16 @@ class ProcessGroupAgent : public RpcAgent {
...
   struct FutureInfo {
     std::shared_ptr<FutureMessage> future_;
-    std::chrono::milliseconds startTime_;
+    std::chrono::milliseconds endTime_;
     int dstRank_;
     std::chrono::milliseconds timeout_;
     FutureInfo(
         const std::shared_ptr<FutureMessage>& future,
-        const std::chrono::milliseconds& startTime,
+        const std::chrono::milliseconds& endTime_,
```
Comment (mrshenli) [inline torch/csrc/distributed/rpc/process_group_agent.h]: remove the trailing underscore from the var name?

**Review by mrshenli — APPROVED** [review]
This LGTM! Just have one minor comment on the var name.

**Review by rohan-varma — APPROVED** [review]
Thanks, this is a great enhancement.

**Review by pietern — CHANGES_REQUESTED** [review]
This piggybacks unrelated changes.

**Comment by xush6528** [comment]
@pietern I re-exported. It's gone.

**Review by rohan-varma — APPROVED** [review]
Restamp since the botched merge has been fixed now.

**Comment by jjlilley** [comment]
fwiw, LGTM, but might need to rebase/reupload, since I believe there was a bad merge earlier. :/

**Pull request closed without merging**